src/test/ui/suggestions/suggest-using-appropriate-keyword-for-struct-and-enum.stderr

```diff
@@ -0,0 +1,32 @@
+error: expected `:`, found `}`
+  --> $DIR/suggest-using-appropriate-keyword-for-struct-and-enum.rs:3:1
+   |
+LL |     Variant
+   |            - expected `:`
+LL | }
+   | ^ unexpected token
+   |
+help: consider using `enum` instead of `struct`
+   |
+LL | pub enum NotStruct {
+   |     ~~~~
```

Comment:
I'm not entirely sure if we should be as trigger happy as proposed here about suggesting a different ADT type. I feel like it is much more likely that an ADT body will contain mistakes compared to user having specified a wrong kind of ADT.

Here it feels like the user was much more likely to just have forgetten to specify a type `: _` for whatever reason. Though I could still see us making a suggestion as proposed here if there was at least one variant with data. That is in cases like these:

```rust
pub struct NotStruct {
    Variant,
    Variant2(u8),
}
```

Reply:
I agree with nagisa's comment. I think that for a case as ambiguous as these we might be better served by pointing at all the relevant places and let the user figure what they wanted. That could be done for cases like this one by adding a label to the `struct` keyword saying "while parsing this `struct`".

Reply:
The comment above is still valid. I would prefer to simplify the logic and "just" add a span_label pointing at the item type being parsed. The likelihood of false positives is too high as it is.